Deposits

MagicalTrev0r opened the pull request.

MagicalTrev0r commented:
class RpcServer : public HttpServer {
       ^~~~~~~~~
/root/chaos/src/Rpc/RpcServer.cpp:508:6: error: no declaration matches ‘bool CryptoNote::RpcServer::on_get_fee_address(const request&, CryptoNote::COMMAND_RPC_GET_FEE_ADDRESS::response&)’
 bool RpcServer::on_get_fee_address(const COMMAND_RPC_GET_FEE_ADDRESS::request& req, COMMAND_RPC_GET_FEE_ADDRESS::response& res) {
      ^~~~~~~~~
/root/chaos/src/Rpc/RpcServer.cpp:5

Menutra commented:
[ 82%] Building CXX object src/CMakeFiles/Wallet.dir/Wallet/PoolRpcServer.cpp.o
In file included from /root/Testnet-p1/src/CryptoNoteCore/CryptoNoteSerialization.h:10,
                 from /root/Testnet-p1/src/CryptoNoteProtocol/CryptoNoteProtocolDefinitions.h:15,
                 from /root/Testnet-p1/src/Wallet/WalletRpcServerCommandsDefinitions.h:8,
                 from /root/Testnet-p1/src/Wallet/PoolRpcServer.h:13,
                 from /root/Testnet-p1/src/Wallet/PoolRpcServer.cpp:8:
/root/Testnet-p1/src/crypto/chacha8.h: In function ‘void Crypto::generate_chacha8_key(Crypto::cn_context&, const string&, Crypto::chacha8_key&)’:
/root/Testnet-p1/src/crypto/chacha8.h:49:40: warning: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of non-trivially copyable type ‘struct Crypto::chacha8_key’; use copy-assignment or copy-initialization instead [-Wclass-memaccess]
     memcpy(&key, &pwd_hash, sizeof(key));
                                        ^
/root/Testnet-p1/src/crypto/chacha8.h:24:10: note: ‘struct Crypto::chacha8_key’ declared here
   struct chacha8_key {
          ^~~~~~~~~~~
In file included from /root/Testnet-p1/src/Wallet/PoolRpcServer.h:14,
                 from /root/Testnet-p1/src/Wallet/PoolRpcServer.cpp:8:
/root/Testnet-p1/src/WalletLegacy/WalletLegacy.h: At global scope:
/root/Testnet-p1/src/WalletLegacy/WalletLegacy.h:103:25: error: ‘virtual CryptoNote::TransactionId CryptoNote::WalletLegacy::getDeposit(uint64_t, uint64_t, uint64_t, uint64_t)’ marked ‘override’, but does not override
   virtual TransactionId getDeposit(uint64_t term, uint64_t amount, uint64_t fee, uint64_t mixIn = 4) override;
                         ^~~~~~~~~~
/root/Testnet-p1/src/Wallet/PoolRpcServer.cpp: In member function ‘bool Tools::pool_rpc_server::on_create_integrated(const Tools::wallet_rpc::COMMAND_RPC_CREATE_INTEGRATED::request&, Tools::wallet_rpc::COMMAND_RPC_CREATE_INTEGRATED::response&)’:
/root/Testnet-p1/src/Wallet/PoolRpcServer.cpp:342:16: warning: unused variable  valid’ [-Wunused-variable]
     const bool valid = CryptoNote::parseAccountAddressString(prefix,
                ^~~~~
src/CMakeFiles/Wallet.dir/build.make:94: recipe for target 'src/CMakeFiles/Wallet.dir/Wallet/PoolRpcServer.cpp.o' failed
make[2]: *** [src/CMakeFiles/Wallet.dir/Wallet/PoolRpcServer.cpp.o] Error 1
CMakeFiles/Makefile2:580: recipe for target 'src/CMakeFiles/Wallet.dir/all' failed
make[1]: *** [src/CMakeFiles/Wallet.dir/all] Error 2
Makefile:148: recipe for target 'all' failed
make: *** [all] Error 2

Hushy-Project committed:
[Cli-Deposits]

Upload current work.
- Working deposits
- Counts current deposits in wallet
- TEST!: withdraw deposit

Hushy-Project committed:
[Refactor]

Merge new deposit file into transfer file
Rename functions for readability
